security/validateMessage.go

```diff
@@ -2,79 +2,162 @@ package security
-		log.Printf("location can not be empty")
+	location := checkStringNotEmpty(data.Location)
+	//name cannot be empty
+	name := checkStringNotEmpty(data.Name)
+
+	return name && creation && location && checkRollCallId(laoId, data.Creation, data.Name, data.ID)
+}
+
+//checkRollCallId check if id is correct  : SHA256('R'||lao_id||creation||name)
+func 	checkRollCallId(laoId string, creation int64, name string, id []byte) bool {
+	var elementsToHashForDataId []string
+	elementsToHashForDataId = append(elementsToHashForDataId, "R", b64.StdEncoding.EncodeToString([]byte(laoId)), strconv.FormatInt(creation, 10), name)
+	hash := HashOfItems(elementsToHashForDataId)
+	if !bytes.Equal(id, hash) {
+		log.Printf("ID of RollCall invalid: %v should be: %v", string(id), string(hash[:]))
 		return false
 	}
 	return true
 }
 
-func PollCreatedIsValid(data message.DataCreatePoll, message message.Message) bool {
+//checkMeetingId check if id is correct  : SHA256('M'||lao_id||creation||name)
+func 	checkMeetingId(laoId string, creation int64, name string, id []byte) bool {
+	var elementsToHashForDataId []string
+	elementsToHashForDataId = append(elementsToHashForDataId, "M", b64.StdEncoding.EncodeToString([]byte(laoId)), strconv.FormatInt(creation, 10), name)
+	hash := HashOfItems(elementsToHashForDataId)
+	if !bytes.Equal(id, hash) {
+		log.Printf("ID of Meeting invalid: %v should be: %v", string(id), string(hash[:]))
+		return false
+	}
 	return true
 }
 
-func RollCallCreatedIsValid(data message.DataCreateRollCall, message message.Message) bool {
-	return true
+//check if id is correct  : SHA256(organizer||creation||name)
+func checkLaoId(organizer []byte, creation int64, name string,id []byte)bool{
+	var elementsToHashForDataId []string
+elementsToHashForDataId = append(elementsToHashForDataId, b64.StdEncoding.EncodeToString(organizer), strconv.FormatInt(creation, 10), name)
+hash := HashOfItems(elementsToHashForDataId)
+if !bytes.Equal(id, hash) {
+log.Printf("ID of createLAO invalid: %v should be: %v", string(id), string(hash[:]))
+return false
+}
+return true
+}
+//RollCallOpenedIsValid tell if a Roll call is valid on opening or reopening
+func RollCallOpenedIsValid(data message.DataOpenRollCall, laoId string, rollCall event.RollCall) bool {
+	//we start after the creation and we end after the start
+	if data.Start < rollCall.Creation {
+		log.Printf("timestamps not logic.Start before creation.")
+		return false
+	}
+	return checkRollCallId(laoId, rollCall.Creation, rollCall.Name, data.ID)
+}
+
+//RollCallClosedIsValid tell if a rollCall timestamps make sense
+func RollCallClosedIsValid(data message.DataCloseRollCall, laoId string, rollCall event.RollCall) bool {
+	//we start after the creation and we end after the start
+	if data.Start < rollCall.Creation || data.End < data.Start {
+		log.Printf("timestamps not logic.Start before creation.")
+		return false
+	}
+	return checkRollCallId(laoId, rollCall.Creation, rollCall.Name, data.ID)
 }
```

Comment:
I think we should also check that the end timestamp is not too far in the future, otherwise it just won't close the roll call am I wrong ? (As I believe that a roll call's state depends only on the timestamps)